build/build.go

```diff
@@ -643,7 +643,16 @@ func Invoke(ctx context.Context, cfg ContainerConfig) error {
 		return errors.Errorf("result must be provided")
 	}
 	c, res := cfg.ResultCtx.Client, cfg.ResultCtx.Res
-	_, err := c.Build(ctx, client.SolveOpt{}, "buildx", func(ctx context.Context, c gateway.Client) (*gateway.Result, error) {
+
+	mainCtx := ctx
```

Comment:
Worth looking into this case in buildkit client later on so hacks like this are not required. It looks like that if this context gets called in a certain timeframe the `/solve` grpc call is canceled before the callback returns, causing some calls inside callback to fail with "no such job". `Build()` should never return if the callback has not returned, even if the grpc already goes down.